include/split_grid.h

```diff
@@ -0,0 +1,205 @@
+#pragma once
+
+#include <quda.h>
+#include <comm_quda.h>
+#include <communicator_quda.h>
+
+#include <gauge_field.h>
+#include <color_spinor_field.h>
+#include <clover_field.h>
+
+int comm_rank_from_coords(const int *coords);
+
+namespace quda
+{
+
+  template <class Field>
+  void inline split_field(Field &collect_field, std::vector<Field *> &v_base_field, const CommKey &comm_key,
+                          QudaPCType pc_type = QUDA_4D_PC)
+  {
+    CommKey comm_grid_dim = {comm_dim(0), comm_dim(1), comm_dim(2), comm_dim(3)};
+    CommKey comm_grid_idx = {comm_coord(0), comm_coord(1), comm_coord(2), comm_coord(3)};
+
+    int rank = comm_rank();
+    int total_rank = product(comm_grid_dim);
+
+    /**
+      The term partition in the variable names and comments can mean two things:
+      - The processor grid (with dimension comm_grid_dim) is divided into (sub)partitions.
+      - For the collecting field, on each processor it contains several partitions, each partition is a copy of
+        the base field.
+      The term partition_dim means the number of partitions in each direction, and (unsurprisingly) partition_dim
+      is the same for the above two meanings, i.e. if I divide the overall processor grid by 3 in one direction,
+      the collect field will be 3 times fatter compared to the base field, in that direction.
+
+      In this file the term *_dim and *_idx are all arrays of 4 int's - one can simplify them as 1d-int to understand
+      things and the extension to 4d is trivial.
+    */
+
+    auto processor_dim = comm_grid_dim / comm_key;      // How many processors are there in a parititon?
```

Comment:
Would be useful to resolve the partition name clash at least int the comments ...